### PR TITLE
add s3 interface

### PR DESCRIPTION
**Pull request by liuqinguestc**
add s3 interface

### CODE REVIEW

File: openapi-spec/swagger.yaml [hunk 1]
```diff
@@ -532,11 +516,254 @@ paths:
...
+      summary: list objects
+      operationId: listObjects
+      responses:
+        '200':
+          description: successful list objects
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ListObjectResponse'
+  'v1/s3/{bucketName}/{object}?uploads':
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: Query strings `uploads`  in paths are not allowed, please modify it.

File: openapi-spec/swagger.yaml [hunk 2]
```diff
@@ -532,11 +516,254 @@ paths:
...
+      summary: upload a object.
+      operationId: init upload
+      responses:
+        '200':
+          description: successful upload a object
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/InitiateMultipartUploadResult'
+  'v1/s3/{bucketName}/{object}?partNumber={PartNumber}&uploadId={UploadId}':
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: Query strings uploads in paths are not allowed

File: openapi-spec/swagger.yaml [hunk 3]
```diff
@@ -532,11 +516,254 @@ paths:
...
+      summary: upload a object part.
+      operationId: uploadpart
+      responses:
+        '200':
+          description: successful upload a part
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/UploadPartResult'
+  'v1/s3/{bucketName}/{object}?uploadId={UploadId}':
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: Query strings uploads in paths are not allowed

File: openapi-spec/swagger.yaml (hunk 1 above)
Comment (leonwanghui) [inline openapi-spec/swagger.yaml]: If I get it right, this should be changed into `v1/s3/{bucketName}/{object}/uploads`.

File: openapi-spec/swagger.yaml (hunk 2 above)
Comment (leonwanghui) [inline openapi-spec/swagger.yaml]: Same comment here

**Pull request closed without merging**